README.md

```diff
@@ -15,6 +15,86 @@
 A modular set of compiler APIs empowering the next generation of Solidity code analysis and developer tooling.
 Written in Rust and distributed in multiple languages.
 
+Slang analyzes Solidity source code and generates a rich concrete syntax tree (CST) that can be reasoned about. This is a departure from the classic approach of "black-box" compilers, which are handed the input and only their output can be observed.
```

Comment:
Once it is approved, let's sync the contents with other `_PRODUCT_README_` markers...